binary/retroplayer update scripts: update to latest tag instead of HEAD

HiassofT opened the pull request:
This avoids picking up unreleased development commits

Also add environment variables to keep cloned git dirs and to bump PKG_REV when PKG_VERSION is bumped. The latter is needed to make sure the currently too new addons will be updated to the correct version.

use eg like this:
```
KEEP_GIT_DIRS=yes BUMP_PKG_REV=yes ./update_binary-addons master
```

MilhouseVH commented on tools/mkpkg/update_binary-addons:
```diff
@@ -152,27 +197,21 @@ for addontxt in $KODI_DIR/cmake/addons/bootstrap/repositories/*-addons.txt ; do
-
-      rm -rf $PKG_NAME.git
+      update_to_latest_tag \
+        ${ROOT}/packages/mediacenter/kodi-binary-addons/${ADDON}/package.mk \
```
```suggestion
        "${ROOT}/packages/mediacenter/kodi-binary-addons/${ADDON}/package.mk" \
```

Just in case `${ROOT}` includes spaces...

HiassofT replied:
will do

HiassofT replied:
oh, there are tons of unquoted more ROOT references, will look at them as well

MilhouseVH commented on tools/mkpkg/update_binary-addons:
```diff
@@ -202,5 +241,7 @@ for ADDON in $(ls -1 ${ROOT}/packages/mediacenter/kodi-binary-addons); do
   echo "Resolving hash for ${ADDON}: HEAD => ${RESOLVED_HASH}"
   update_pkg ${ROOT}/packages/mediacenter/kodi-binary-addons/$ADDON/package.mk ${ADDON} ${RESOLVED_HASH}
```
```suggestion
  update_pkg "${ROOT}/packages/mediacenter/kodi-binary-addons/$ADDON/package.mk" ${ADDON} ${RESOLVED_HASH}
```

Same.

HiassofT commented on tools/mkpkg/update_binary-addons:
```diff
@@ -105,6 +116,36 @@ update_pkg() {
+  resolved_version=$(resolve_tag ${gitdir} ${ref})
+  if [ -z "${resolved_version}" ]; then
+    echo "========================================================================"
+    echo "ERROR: no tag found for package ${pkg_name}, falling back to HEAD"
```
thanks, makes sense

HiassofT commented:
I've pushed my update script refactoring (moving common stuff into update_common_functions)
and also added an experimental commit to switch update_retroplayer-addons to using tags.

Better don't use the latter in production ATM as tags in the kodi-game repos are somewhat out of date

MilhouseVH commented on tools/mkpkg/update_binary-addons:
```diff
@@ -9,104 +9,23 @@ if [ -z "$1" ]; then
-  local addon="$1" url1="$2"
-  local domain owner repo url2
+MY_DIR=$(dirname $0)
+ROOT=$(cd "${MY_DIR}"/../.. && pwd)
```
```
MY_DIR="$(dirname $0)"
ROOT="$(cd "${MY_DIR}"/../.. && pwd)"
```
?

HiassofT replied:
thanks, yes, and $0 needs to be quoted as well

MilhouseVH commented on tools/mkpkg/update_retroplayer-addons:
```diff
@@ -10,101 +10,19 @@ if [ "$1" == "-f" ]; then
 
-ROOT=$(cd $(dirname $0)/../.. && pwd)
+MY_DIR=$(dirname $0)
+ROOT=$(cd "${MY_DIR}"/../.. && pwd)
```
```
MY_DIR="$(dirname $0)"
ROOT="$(cd "${MY_DIR}"/../.. && pwd)"
```

HiassofT commented:
thanks for the review, I fixed the path quoting, the update scripts now run fine in a directory with spaces.

tested with
```
hias@lenny:~/private/libreelec/test dir$ KEEP_GIT_DIRS=yes BUMP_PKG_REV=yes ../libreelec-git/tools/mkpkg/update_retroplayer-addons
```

having the LE tree in a directory with spaces won't work though, sourcing config/options doesn't succeed in that case.

HiassofT commented:
I've added options to allow excluding packages from updates so we can drop vfs.sacd and game.netplay without being constantly nagged that they are present in upstream binary repo but missing in LE.

I also added commits to remove vfs.sacd, which is currently unreleased upstream, and game.netplay.

I've kept peripheral.steamcontroller for now, it's not tagged upstream and most certainly not needed on newer kernels which support the steam controller, but OTOH it's still receiving updates upstream and may be useful on RK and AML which run older kernels.

HiassofT commented:
addon updates now look fine, here's a tree with bumps to latest tags: https://github.com/HiassofT/LibreELEC.tv/commits/le92-binary-addon-bump

Kodi game addons seem to be OK, too, game.libretro tag is correct (sorry, looked at master instead of retroplayer branch yesterday...) and the game.libretro.xxx commits after the tags seem to be buildsystem changes not needed in LE.

HiassofT commented:
I've squashed the minor fixes and added the missing imagedecoder.mpo addon (whatever that is), which build-tested fine for RPi2.

I also updated the tree with binary/game addon updates: https://github.com/HiassofT/LibreELEC.tv/commits/le92-binary-addon-bump, build test with create_addon all is currently running

HiassofT commented:
RPi2 image build and `scripts/create_addon all` of addon update tree was fine
`ALL ADDONS BUILT SUCCESSFULLY`
